src/main/java/io/stargate/sgv2/jsonapi/service/operation/reranking/FeatureUsage.java

```diff
@@ -0,0 +1,125 @@
+package io.stargate.sgv2.jsonapi.service.operation.reranking;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a collection of {@link Feature}s in a command. This class is immutable and can be used
+ * to track which features are utilized in a command. This class is immutable; methods that modify
+ * the set of features return a new {@code FeatureUsage} instance. It uses an {@link EnumSet}
+ * internally for efficient storage and operations on features.
+ */
+public final class FeatureUsage {
```

Comment:
If we renamed `Feature` as `CommandFeature`, then this should probably become just `CommandFeatures` (without "usage").